packages/grid_client/scripts/applications/nostor.ts

```diff
@@ -0,0 +1,137 @@
+import { log, pingNodes } from "../utils";
+
+async function deploy(client: GridClient, vms: MachinesModel, subdomain: string, gatewayNode: any) {
+  // Deploy VM
+  const resultVM = await client.machines.deploy(vms);
+  log("================= Deploying VM =================");
+  log(resultVM);
+  log("================= Deploying VM =================");
+
+  // Get WG interface details
+  const wgnet = (await client.machines.getObj(vms.name))[0].interfaces[0];
+
+  // Deploy Gateway
+  const gateway: GatewayNameModel = {
+    name: subdomain,
+    network: wgnet.network,
+    node_id: gatewayNode.nodeId,
+    tls_passthrough: false,
+    backends: [`http://${wgnet.ip}:8080`],
+  };
+
+  const resultGateway = await client.gateway.deploy_name(gateway);
+  log("================= Deploying Gateway =================");
+  log(resultGateway);
+  log("================= Deploying Gateway =================");
+}
+
+async function getDeployment(client: GridClient, name: string, subdomain: string) {
+  // Get VM deployment
+  const resultVM = await client.machines.getObj(name);
+  log("================= Getting VM Deployment =================");
+  log(resultVM);
+  log("================= Getting VM Deployment =================");
+
+  // Get Gateway deployment
+  const resultGateway = await client.gateway.getObj(subdomain);
+  log("================= Getting Gateway Deployment =================");
+  log(resultGateway);
+  log(`https://${resultGateway[0].domain}`);
+  log("================= Getting Gateway Deployment =================");
+}
+
+async function cancel(client: GridClient, name: string, subdomain: string) {
+  // Cancel VM deployment
+  const resultVM = await client.machines.delete({ name });
+  log("================= Canceling VM Deployment =================");
+  log(resultVM);
+  log("================= Canceling VM Deployment =================");
+
+  // Cancel Gateway deployment
+  const resultGateway = await client.gateway.delete_name({ name: subdomain });
+  log("================= Canceling Gateway Deployment =================");
+  log(resultGateway);
+  log("================= Canceling Gateway Deployment =================");
+}
+
+async function main() {
+  const name = "newnostr1";
+  const grid3 = await getClient(`nostr/${name}`);
+  const subdomain = `ntt${grid3.twinId}${name}`;
+  const instanceCapacity = { cru: 2, mru: 4, sru: 50 };
+
+  // VM Query Options
+  const vmQueryOptions: FilterOptions = {
+    features: [Features.wireguard, Features.mycelium],
+    cru: instanceCapacity.cru,
+    mru: instanceCapacity.mru,
+    sru: instanceCapacity.sru,
+    availableFor: grid3.twinId,
+    farmId: 1,
+  };
+
+  // Gateway Query Options
+  const gatewayQueryOptions: FilterOptions = {
+    features: [Features.wireguard, Features.mycelium, Features.gatewaynameproxy],
+    gateway: true,
+    availableFor: grid3.twinId,
+  };
+
+  const gatewayNode = (await grid3.capacity.filterNodes(gatewayQueryOptions))[0];
```

Comment:
we should use the ping node here to same as the vm node; get the full resopvense and pass it to ping node

Reply:
could not enhance this without the script to fail